### PR TITLE
fix #9320 - UDF functions use own database

### LINKED ISSUE #9320: Test: UdfDbFunctionSqlServerTests class should not use northwind database
`UdfDbFunctionSqlServerTests` is used for functional testing for UDFs. Any time we change something and want to add more test that should be the class to go to. But at present that class uses northwind database. 
We don't want to modify northwind database since anytime we modify the script, we need to drop database from machines where it is persisted. (CI & local dev boxes).
We should let this class use northwind model (or may be new model) but use own database which would get recreated every time so that adding more UDFs don't require manual dropping of database on various machines.

This should also remove currently defined UDFs in northwind.

### PR DESCRIPTION
**Pull request by pmiddleton**
I have updated the Sql Server UDF tests to use their own database/model instead of Northwind.

The model is very similar to the subset of Northwind which was being used previously.  Employee is now Customer and less data is loaded.

All the tests are structurally the same.

### CODE REVIEW

File: test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs [hunk 1]
```diff
@@ -2,47 +2,245 @@
...
+                InitalizeDatabase(context);
+
+                ClearLog();
+            }
         }
 
-        protected NorthwindQuerySqlServerFixture<NoopModelCustomizer> Fixture { get; }
+        private void CreateDbFunctions(UDFSqlContext context)
+        {
+            context.Database.ExecuteSqlCommand(@"create function [dbo].[CustomerOrderCount] (@customerId int)
```
Comment (AndriySvyryd) [inline test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs]: Add a new fixture derived from `SharedStoreFixtureBase<DbContext>` and create the functions in `Seed`  method, see `SqlServerTriggersTest`

**Comment by AndriySvyryd** [comment]
Don't modify `Northwind.sql` at this point it's better to just leaving it as is

**Comment by smitpatel** [comment]
@AndriySvyryd - Let's discuss tomorrow in team room about modifying northwind. It would again breaking change for dev boxes but removing the UDF functions from northwind allows us to keep the databse in sync across multiple providers running the tests.

**Comment by pmiddleton** [comment]
I don't see why this would be a breaking change.  The functions will still be on your local instance, but it won't cause any issues since they won't be used anymore by the unit tests.  

Let me know what is decided in your meeting.

**Comment by AndriySvyryd** [comment]
@pmiddleton We've discussed this and removing the functions from `Northwind.sql` is fine. It will cause minor issues when working on a patch branch, but will be best in the long-term.

File: test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs (hunk 1 above)
Comment (pmiddleton) [reply test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs]: I don't see a way to access the generated sql with this fixture.  Ideas on what I can use to get access to the sql?

Comment (AndriySvyryd) [reply test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs]: Add this property to the fixture:
```C#
public TestSqlLoggerFactory TestSqlLoggerFactory => (TestSqlLoggerFactory)ServiceProvider.GetRequiredService<ILoggerFactory>();
```

Comment (pmiddleton) [reply test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs]: I figured out how to get access to the TestSqlLoggerFactory.  

Should we expose an AssertSql and ClearLog method on the SharedStoreFixtureBase?

Comment (AndriySvyryd) [reply test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs]: We cannot do that as `SharedStoreFixtureBase` is at the Core level, so it is also used for non-relational provider tests. We'll consider adding a `RelationalSharedStoreFixtureBase` if this keeps coming up.

File: test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs [hunk 2]
```diff
@@ -2,47 +2,248 @@
...
+            public void ClearLog()
+                => ((TestSqlLoggerFactory)ServiceProvider.GetRequiredService<ILoggerFactory>()).Clear();
+
+            public void AssertSql(params string[] expected)
+                => ((TestSqlLoggerFactory)ServiceProvider.GetRequiredService<ILoggerFactory>()).AssertBaseline(expected);
+
+            public override DbContextOptionsBuilder AddOptions(DbContextOptionsBuilder builder)
+            {
+                base.AddOptions(builder);
+                return builder.ConfigureWarnings(w => w.Default(WarningBehavior.Ignore));
```
Comment (AndriySvyryd) [inline test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs]: Consider just ignoring specific warnings

File: test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs [hunk 3]
```diff
@@ -2,47 +2,248 @@
...
+        public UdfDbFunctionSqlServerTests(SqlServerUDFFixture fixture)
         {
             Fixture = fixture;
+            Fixture.ClearLog();
+        }
+
+        private SqlServerUDFFixture Fixture { get; }
+
+        public class SqlServerUDFFixture : SharedStoreFixtureBase<DbContext>
+        {
```
Comment (smitpatel) [inline test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs]: Fixture goes to bottom of the file, no? @AndriySvyryd 
e.g. https://github.com/aspnet/EntityFrameworkCore/blob/dev/test/EFCore.SqlServer.FunctionalTests/Query/IncludeOneToOneSqlServerTest.cs

Comment (AndriySvyryd) [reply test/EFCore.SqlServer.FunctionalTests/Query/UdfDbFunctionSqlServerTests.cs]: Yes

**Comment by AndriySvyryd** [comment]
@pmiddleton Thanks for your contribution!

**Pull request merged**